docs/reusing-typecheck-results.md

```diff
@@ -0,0 +1,317 @@
+---
+title: Reusing compilation results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing compilation results between compiler and tooling runs
+
+Caching and reusing compilation results between compiler and tooling runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+## Motivation
+
+- **Performance**. Things like recomputing type information for large projects are time-consuming. By caching the results, subsequent compilation can skip some steps, leading to faster builds.
+
+- **Better IDE cold start**. IDEs can provide faster IntelliSense, code navigation, and other features if they can quickly access cached compilation information.
+
+- **Error reduction**. Consistent compilation results help avoid scenarios where slight differences between runs could cause different errors or warnings to be produced.
```

Comment:
What are the specific examples?
How would the feature help at them?


Without real examples, it would be better to have this bullet point excluded to avoid confusion of what this can really bring.

Reply:
Removed also